core-processor/src/processor.rs

```diff
@@ -233,87 +231,192 @@ pub fn precharge(
     }
 }
 
-/// Prepares environment for the execution of a program.
-/// Checks if there is a required export and tries to charge for code and memory pages.
-/// Returns either `PreparedMessageExecutionContext` for `process` or an array of journal notes.
-/// See `PrepareResult` for details.
-pub fn prepare(
+/// Charge a message for fetching the length of the binary code.
+pub fn precharge_for_code_length(
```

Comment:
This name is misleading. And comment also. 
`precharge_for_code_length` is more suitable for `precharge_for_code` and vice versa

Reply:
Frankly speaking you confused me =).

Length of an instrumented binary code is retained in another new storage after changes introduced by this PR. This is required since after reinstrumentation the processed code could have different length than its previous version. To get this very value of the length we have to charge for onetime read operation from DB: that is rendered in the function name.

However I agree with you that comment is not much informative since the function performs additional checking. Will edit it to make it more elaborate.